cmake/thirdparty/get_legion.cmake

```diff
@@ -0,0 +1,193 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#=============================================================================
+
+function(find_or_configure_legion)
+  set(oneValueArgs VERSION REPOSITORY BRANCH EXCLUDE_FROM_ALL)
+  cmake_parse_arguments(PKG "${options}" "${oneValueArgs}" "${multiValueArgs}" ${ARGN})
+
+  include("${rapids-cmake-dir}/export/detail/parse_version.cmake")
+  rapids_export_parse_version(${PKG_VERSION} Legion PKG_VERSION)
+
+  set(Legion_CUDA_ARCH "")
+  if(Legion_USE_CUDA)
+    set(Legion_CUDA_ARCH ${CMAKE_CUDA_ARCHITECTURES})
+    list(TRANSFORM Legion_CUDA_ARCH REPLACE "-real" "")
+    list(TRANSFORM Legion_CUDA_ARCH REPLACE "-virtual" "")
+  endif()
+
+  set(FIND_PKG_ARGS
+      GLOBAL_TARGETS     Legion::Realm
+                         Legion::Regent
+                         Legion::Legion
+                         Legion::RealmRuntime
+                         Legion::LegionRuntime
+      BUILD_EXPORT_SET   legate-core-exports
+      INSTALL_EXPORT_SET legate-core-exports)
+
+  # First try to find Legion via find_package()
+  # so the `Legion_USE_*` variables are visible
+  # Use QUIET find by default.
+  set(_find_mode QUIET)
+  # If Legion_DIR/Legion_ROOT are defined as something other than empty or NOTFOUND
+  # use a REQUIRED find so that the build does not silently download Legion.
+  if(Legion_DIR OR Legion_ROOT)
+    set(_find_mode REQUIRED)
+  endif()
+  rapids_find_package(Legion ${PKG_VERSION} EXACT CONFIG ${_find_mode} ${FIND_PKG_ARGS})
+
+  if(Legion_FOUND)
+    message(STATUS "CPM: using local package Legion@${PKG_VERSION}")
+  else()
+    include(${CMAKE_CURRENT_SOURCE_DIR}/cmake/Modules/cpm_helpers.cmake)
+    get_cpm_git_args(legion_cpm_git_args REPOSITORY ${PKG_REPOSITORY} BRANCH ${PKG_BRANCH})
+    if(NOT DEFINED Legion_PYTHON_EXTRA_INSTALL_ARGS)
+      set(Legion_PYTHON_EXTRA_INSTALL_ARGS "--single-version-externally-managed --root=/")
+    endif()
+
+    set(_cuda_path "")
+    set(_legion_cuda_options "")
+
+    # Set CMAKE_CXX_STANDARD and CMAKE_CUDA_STANDARD for Legion builds. Legion's FindCUDA.cmake
+    # use causes CUDA object compilation to fail if `-std=` flag is present in `CXXFLAGS` but
+    # missing in `CUDA_NVCC_FLAGS`.
+    set(_cxx_std "${CMAKE_CXX_STANDARD}")
+    if(NOT _cxx_std)
+      set(_cxx_std 17)
+    endif()
+
+    if(Legion_USE_CUDA)
+      set(_cuda_std "${CMAKE_CUDA_STANDARD}")
+      if(NOT _cuda_std)
+        set(_cuda_std ${_cxx_std})
+      endif()
+
+      if(NOT CUDA_NVCC_FLAGS)
+        list(APPEND CUDA_NVCC_FLAGS "${CUDAFLAGS}")
+      endif()
+
+      set(_nvcc_flags ${CUDA_NVCC_FLAGS})
+      if(NOT "${_nvcc_flags}" MATCHES "-std=")
+        list(APPEND _nvcc_flags "-std=c++${_cuda_std}")
+      endif()
+
+      # Get the `stubs/libcuda.so` path so we can set CMAKE_LIBRARY_PATH for FindCUDA.cmake
+
+      # Prefer users' CUDA_PATH envvar (if set)
+      set(_cuda_stubs "$ENV{CUDA_PATH}")
```

Comment:
Can we check in all cases that `<candidate-dir>/stubs/libcuda.so` exists before accepting it? e.g. something like
```
does ENV{CUDA_PATH}/stubs/libcuda.so exist?
if yes then _cuda_stubs += ENV{CUDA_PATH}/stubs otherwise:
does ENV{CUDA_LIB_PATH}/stubs/libcuda.so exist?
if yes then _cuda_stubs += ENV{CUDA_LIB_PATH}/stubs otherwise:
...
```